core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1217,6 +1217,66 @@ public final This bindList(String key, List<?> values) {
     }
 
     /**
+     * Bind a parameter for each value in the given list * number of property names,
+     * and defines an attribute as the comma-separated list of parameter references (using colon prefix).
+     *
+     * Used to create query similar to:
+     * select * from things where (id, foo) in ((1,'abc'),(2,'def'),(3,'ghi'))
+     * <p>
+     * Examples:
+     * <pre>
+     *
+     * List&lt;Things&gt; thingsToSearchBy = ...
+     * List&lt;Thing&gt; things = handle.createQuery("select * from things where (id, foo) in (&lt;thingKeys&gt;)")
```

Comment:
I'm not sure what you mean about defining 2 keys. Can you give me an example on how you envision it looking?

I agree that it's a bit of duplication to define the keys twice. As you mentioned, that's hard to solve if the bean properties don't match the DB column names. At the very least, I think the current API on the branch is required for that use case. Could definitely extend the API to make the code more DRY in the case where bean properties match the DB column names though.